app/components/viewer/preload.js

```diff
@@ -1,21 +1,261 @@
 const { ipcRenderer } = require('electron')
 
-const callbacks = []
+let currentId = 0.0
 
-window.read = (start, end, cb) =>
-  send('read', [start, end], (err, str) => {
-    if (err) return cb(err)
-    cb(null, Buffer.from(str, 'hex'))
+/*
+;['log', 'warn', 'error', 'info'].forEach(function (prop) {
+  const old = console[prop]
+  console[prop] = function () {
+    let arg = arguments[0]
+    if (arguments.length > 1) {
+      arg = Array.from(arguments).join(' ')
+    }
+    old.call(console, arg)
+  }
+})
+*/
```

Comment:
Logging of errors - right now - doesn't provide a stack trace but at least keeps the last line-number for some debugging. However, the current logging does not take care of the additional arguments, so the block above concats the arguments. We loose the stack of the error though :(

Reply:
error {de,}serialization is often a pain, this might be helped by using a module like https://www.npmjs.com/package/serialize-error

Reply:
Looks interesting, ... https://www.npmjs.com/package/serialize-error